app/views/content_providers/show.html.erb

```diff
@@ -64,7 +64,7 @@
                      locals: { total_count: upcoming_events.count,
                                past_count: past_events.count,
                                limit: resource_limit,
-                               resources: events,
+                               resources: events.sort_by(&:start),
```

Comment:
Better to order `upcoming_events` with `.order(start: :asc)`, otherwise you are just sorting the set of 30 events that were returned.